PR for submission of the paper

profxj opened the pull request:
As titled!

profxj committed:
wip

profxj committed:
analysis runnin

profxj committed:
mo

profxj committed:
refactored

profxj committed:
mo

profxj committed:
mo

profxj committed:
wip

profxj committed:
polish

profxj committed:
workin

profxj committed:
mo

profxj committed:
first results

profxj committed:
fig1

profxj committed:
mo

profxj committed:
fig 2

profxj committed:
cdom

profxj committed:
cdom

profxj committed:
wip

profxj committed:
wip

profxj committed:
fit chl

profxj committed:
cut

profxj committed:
mo

profxj committed:
coeff degenearcy

profxj committed:
mo

profxj committed:
first try

profxj committed:
this and that

profxj committed:
mo

profxj committed:
this and that

profxj committed:
nb

profxj committed:
wip

profxj committed:
wip

profxj committed:
fig 1

profxj committed:
fig 1

profxj committed:
fig 3

profxj committed:
fig 5

profxj committed:
fig 6?

profxj committed:
notebooks

profxj committed:
lots wip

profxj committed:
wip

profxj committed:
wip

profxj committed:
high

profxj committed:
mystery bug

profxj committed:
mo

profxj committed:
normalize

profxj committed:
jxp

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
jxp

profxj committed:
wip

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
still fussin

profxj committed:
figs

profxj committed:
no bug?

profxj committed:
minor

profxj committed:
refactoring..

profxj committed:
mo

profxj committed:
mo

profxj committed:
N=3

profxj committed:
label

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
wip

profxj committed:
M

profxj committed:
coeffs

profxj committed:
fussin

profxj committed:
figs

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
exploring chl-a

profxj committed:
oskar

profxj committed:
figs

profxj committed:
this and that

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
nmf draft

profxj committed:
mo

profxj committed:
Tara

profxj committed:
NAP

profxj committed:
Chl

profxj committed:
mo

profxj committed:
fussin

profxj committed:
phyto

profxj committed:
edits

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
tara

profxj committed:
wip

profxj committed:
new fit

profxj committed:
progress

profxj committed:
fig

profxj committed:
water and figs

profxj committed:
variance

profxj committed:
explore

profxj committed:
figs and figs

profxj committed:
tara figs

profxj committed:
more

profxj committed:
mo figs

profxj committed:
outliers

profxj committed:
conflicts coming

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
outliers

profxj committed:
mo

profxj committed:
figs

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
mo

profxj committed:
aph

profxj committed:
mo aph

profxj committed:
figs

profxj committed:
w1

profxj committed:
revamping the figs

profxj committed:
mo

profxj committed:
figs continue

profxj committed:
lots o figure updates

profxj committed:
outliers

profxj committed:
order fixed

profxj committed:
apply

profxj committed:
wip

profxj committed:
fitting

profxj committed:
NMF too

profxj committed:
fit

profxj committed:
bricaud RMSE

profxj committed:
appendix

profxj committed:
mo

profxj committed:
cdom

profxj committed:
nmf updates

profxj committed:
MTRI

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
h2h4

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
nearly ready to submit

profxj committed:
mo

profxj committed:
Merge branch 'paper' of https://github.com/AI-for-Ocean-Science/cnmf into paper

profxj committed:
bricaud fig

profxj committed:
Ren info

profxj committed:
sync up with IHOP

profxj committed:
another update

profxj committed:
table

patrickcgray commented:
Okay a lot here and I'm not done reviewing. We can chat about this tomorrow if you'd like. Primarily environment and data linkage issues that need to be resolved:

Need to note in the README that the ocean-color repo is necessary, and specifically the https://github.com/AI-for-Ocean-Science/ocean-color/tree/nmf_paper_support branch of that repo. But that repo isn't quite deployment ready and files are not included. Could make a release and add them as binaries on the release. 

Then I ran into some issues because various libraries installed as older version that what were than required. For example importlib must be >= 3.9. Easiest would be to just output the versions you're using now as a stable env.

Possibly overkill but I just pulled docker run -it -v /Users/patrickgray:/home/jovyan --rm -p 8888:8888 pangeo/pangeo-notebook:2024.04.05  jupyter lab --ip 0.0.0.0 and then ran `$ pip install .` from the base of the repo which got most everything working.

I had to add the below code at the beginning of all files to get that to run. Could just include in the repo instructions to set these env variables:

```
import sys
import os
sys.path.insert(1, '/home/jovyan/Code/ocean-color')
os.environ["OS_COLOR"] = '/home/jovyan/Code/ocean-color/files'
```

Needed to add the Hydrolight400.nc file at /home/jovyan/Code/ocean-color/files/data/Loisel2023. but can put this wherever we point to it via OS_COLOR.

Within cnmf/papers/I/Analysis/Tara_phyto.ipynb I am unable to get plots to show because the itneractive plotting isn't working for me (i.e. I cannot run `%matplotlib ipympl`. Not a major issue but could be resolve by a specific image or environment file specifying versions. This is likely on my end.

I also get errors where we access the UID attribute from the tara geofeather file. I saved it as `uid` and you have it as `UID` - just something to keep straight. Sometimes `uid` is used. 

I believe we filtered the Tara Microbiome dataset slightly differently, there are some UIDs in your Tara dataset that I don't have in the NMF coeffs dataset (my version must be old). This causes lines such as`midx = cat_utils.match_ids(d['UID'], tara_db.uid.values)` within Tara_chl_outliers.ipynb and Tara_Chl.ipynb and other notebooks to fail.

The function rebin_to_grid() appears to be missing from oceancolor/utils/spectra.py or is in a different branch. This prevents me from running the call to `iops.tara_matched_to_l23(low_cut=410.)` within NMFTara.ipynb .

My file Tara_NMF_04_a.npz appears to be a different size than yours. I have N=185585 and you have N=239880. I assume mine is just out of date. But good to include a definitive version with the release.

Within Outliers_Tara.ipynb the last block breaks because the idx is from Tara and much larger than the L23 dataset. Need to explicitly set this to be < 3320.

Need to add the 3 mode NMF fit data if we want to include notebooks like Phytoplankton.ipynb. Also needs the `data_moore.mat`

Need the L23_NMF_04_a.npz data included to run a few notebooks. Also L23_NMF_02_aph.npz for the Bricaud comparison.

These should mostly be simple fixes. Primarily before merging I would like to re-run the code from NMFTara.ipynb just to ensure what I imagine will be the aspect most people want to use can run easily and consistently. To do this we need the function `rebin_to_grid()` which I think must be in a different branch or just a more up to date local version you have of the `ocean-color` branch.

For data I recommend we include instructions in the README of where to add the data and we can include it either as a Releases binary or add it to some official data repo. I'm ambivalent to which. For now if you want to just throw all the official versions on Google Drive or something so I can test with them.

profxj committed:
bad bug fix

profxj commented:
1.  Updated the README extensively
2.  Removed packages from `setup.py` but did not specify version numbers on any. Can you tell me which ones you had to update?
3.  Added a few packages to `setup.py`, e.g. `ipywidgets` for the Notebook error issue you had
4.  Are we allowed to put the Tara file in Dryad?
5.  Added the L23_XXX.npz files
6. And the m=3 file
7. I'll add more in the README on where to get the Tara data once we settle on that

profxj committed:
edits for PCG

profxj committed:
Merge remote-tracking branch 'origin/refactor_ihop' into paper
